### DISCUSSION TITLE
OpenDDS subscription and publisher in same application to send and receive different messages

### DISCUSSION THREAD

**Discussion by pmaskale** [OpenDDS/OpenDDS#3536] [Q&A] [answered]
What is the best approach to have single application as subscriber and publisher both inside it? Does it have to be same domain participant?
How can we subscriber to multiple messages for same application.

**Comment by jrw972** [comment, marked as answer]
DDS offers a great deal of flexibility in this regard.  An application can can multiple participants.  A participant can have multiple publishers and subscribers.  A publisher can have multiple data writers.  A subscriber can have multiple data readers.  I would recommend starting with a single participant, single publisher, single subscriber until you have a good reason to do something else.

**Reply by pmaskale** [reply]
Thank you for the answer! But all i have to understand is 
1. how to read multiple topics with different typename in a single data reader implementation? 
     [Edit] : I was able to do it.
2. how can the same participant act as publisher for message1 and subscriber for message2? (consider message1 and message2 are different structs using idl).

And yes i have worked with single message for publisher and subscriber it works just fine.

**Reply by jrw972** [reply]
For (2), you need to add two different type supports so that you can create the two different topics.

**Reply by pmaskale** [reply]
Thanks i was able to do that as well!
